test/python/tests/csi/test_csi.py

```diff
@@ -1,83 +1,43 @@
-import logging
 import os
 import pytest
-import signal
 import subprocess
-import threading
 import time
 
+from common.mayastor import container_mod
 from common.hdl import MayastorHandle
 from common.csi_hdl import CsiHandle
 
 import grpc
 import csi_pb2 as pb
 import csi_pb2_grpc as rpc
 
-pytest_plugins = ["docker_compose"]
-
 
 def get_uuid(n):
     return "11111111-0000-0000-0000-%.12d" % (n)
 
 
 @pytest.fixture(scope="module")
-def start_csi_plugin():
-    def monitor(proc, result):
-        stdout, stderr = proc.communicate()
-        result["stdout"] = stdout.decode()
-        result["stderr"] = stderr.decode()
-        result["status"] = proc.returncode
-
-    proc = subprocess.Popen(
-        args=[
-            "sudo",
-            os.environ["SRCDIR"] + "/target/debug/mayastor-csi",
-            "--csi-socket=/tmp/csi.sock",
-            "--grpc-endpoint=0.0.0.0",
-            "--node-name=msn-test",
-            "-v",
-        ],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-    )
-
-    result = {}
-    handler = threading.Thread(target=monitor, args=[proc, result])
-    handler.start()
-    time.sleep(1)
-    yield
-    subprocess.run(["sudo", "pkill", "mayastor-csi"], check=True)
-    handler.join()
-    print("[CSI] exit status: %d" % (result["status"]))
-    print(result["stdout"])
-    print(result["stderr"])
-
-
-@pytest.fixture(scope="module")
-def handles_mod(docker_project, module_scoped_container_getter):
-    assert "ms0" in docker_project.service_names
-    handles = {}
-    handles["ms0"] = MayastorHandle(
-        module_scoped_container_getter.get("ms0").get(
-            "NetworkSettings.Networks.mayastor_net.IPAddress"
-        )
+def mayastor_instance(container_mod):
+    container = container_mod.get("ms0")
+    yield MayastorHandle(
+        container.get("NetworkSettings.Networks.mayastor_net.IPAddress")
     )
-    yield handles
 
 
 @pytest.fixture(scope="module")
-def mayastor_instance(handles_mod):
-    yield handles_mod["ms0"]
+def csi_container(container_mod):
```

Comment:
The function was changed to not return the container and return 'NetworkSettings.Networks.mayastor_net.IPAddress' instead. In case I overlooked it: who/what is the primary consumer of this new information in test modules ?
The only thing I saw was 'fix_socket_permissions', but it doesn't rely on fixture's output in any fashion.

Reply:
See above comment, It is used to enforce a dependency for timing purposes. We need to know that the csi_container is running before we start messing with the socket. As noted above shall change so that it does not produce a value.

Reply:
Thanks.